src/Umbraco.Web/Templates/TemplateUtilities.cs

```diff
@@ -236,6 +244,21 @@ internal static string FindAndPersistPastedTempImages(string html, int mediaPare
 
                 // Remove the data attribute (so we do not re-process this)
                 img.Attributes.Remove(TemporaryImageDataAttribute);
+
+                // Delete folder & image now its saved in media
+                // The folder should contain one image - as a unique guid folder created
+                // for each image uploaded from TinyMceController
+                var folderName = Path.GetDirectoryName(absoluteTempImagePath);
+                try
+                {
+                    File.Delete(absoluteTempImagePath);
+                    Directory.Delete(folderName);
```

Comment:
```suggestion
                    Directory.Delete(folderName, true);
```

If you do this, you can skip the explicit `File.Delete(absoluteTempImagePath);`

Reply:
Thanks, I can never remember. Always thought IO would moan & complain if the folder/directory was not empty